examples/basic/gpu_shared.c

```diff
@@ -0,0 +1,191 @@
+    assert(ze_result == ZE_RESULT_SUCCESS);
+    assert(*context != NULL);
+
+    // Free the memory
+    free(all_drivers);
+    return 0;
+}
+
+int main(void) {
+
+    // A result object for storing UMF API result status
+    umf_result_t res;
+
+    // Initialize Level Zero
+    ze_driver_handle_t hDriver;
+    ze_device_handle_t hDevice;
+    ze_context_handle_t hContext;
+    int ret = level_zero_init(&hDriver, &hDevice, &hContext);
+    if (ret < 0) {
+        // Initialization failed or no GPU devices was found
+        return -1;
+    }
+
+    // Setup parameters for the Level Zero memory provider. It will be used for
+    // allocating memory from Level Zero devices.
+    level_zero_memory_provider_params_t ze_memory_provider_params;
+    ze_memory_provider_params.level_zero_context_handle = hContext;
+    ze_memory_provider_params.level_zero_device_handle = hDevice;
+    // Set the memory type to shared to allow the memory to be accessed on both
+    // CPU and GPU.
+    ze_memory_provider_params.memory_type = UMF_MEMORY_TYPE_SHARED;
+
+    // Create Level Zero memory provider
+    umf_memory_provider_handle_t ze_memory_provider;
+    res = umfMemoryProviderCreate(umfLevelZeroMemoryProviderOps(),
+                                  &ze_memory_provider_params,
+                                  &ze_memory_provider);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to create a memory provider!");
+        goto level_zero_destroy;
+    }
+
+    printf("Level Zero memory provider created at %p\n",
+           (void *)ze_memory_provider);
+
+    // Setup parameters for the Disjoint Pool. It will be used for managing the
+    // memory allocated using memory provider.
+    umf_disjoint_pool_params_t disjoint_memory_pool_params =
+        umfDisjointPoolParamsDefault();
+    // Set the Slab Min Size to 64KB - the page size for GPU allocations
+    disjoint_memory_pool_params.SlabMinSize = 64 * 1024L;
+    // We would keep only single slab per each allocation bucket
+    disjoint_memory_pool_params.Capacity = 1;
+    // Set the maximum poolable size to 64KB - objects with size above this
+    // limit will not be stored/allocated from the pool.
+    disjoint_memory_pool_params.MaxPoolableSize = 64 * 1024L;
+    // Enable tracing
+    disjoint_memory_pool_params.PoolTrace = 1;
+
+    // Create Disjoint Pool memory pool.
+    umf_memory_pool_handle_t ze_disjoint_memory_pool;
+    res = umfPoolCreate(umfDisjointPoolOps(), ze_memory_provider,
+                        &disjoint_memory_pool_params, UMF_POOL_CREATE_FLAG_NONE,
+                        &ze_disjoint_memory_pool);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to create a memory pool!");
+        goto memory_provider_destroy;
+    }
+
+    printf("Disjoint Pool created at %p\n", (void *)ze_disjoint_memory_pool);
+
+    // Allocate some memory from the pool
+    void *ptr = umfPoolMalloc(ze_disjoint_memory_pool, sizeof(int));
```

Comment:
int *ptr 

and then remove cast from the following line

Reply:
done